packages/select/src/Select/Select.tsx

```diff
@@ -0,0 +1,121 @@
+import { Combobox } from "@telegraph/combobox";
+import { TgphComponentProps } from "@telegraph/helpers";
+import React from "react";
+
+type Option = TgphComponentProps<typeof Combobox.Option>;
+
+type SingleValue = string;
+type MultipleValue = Array<string>;
+
+type RootProps = Omit<
+  TgphComponentProps<typeof Combobox.Root>,
+  "value" | "onValueChange" | "layout"
+> & {
+  size?: TgphComponentProps<typeof Combobox.Trigger>["size"];
+  triggerProps?: TgphComponentProps<typeof Combobox.Trigger>;
+  contentProps?: TgphComponentProps<typeof Combobox.Content>;
+  optionsProps?: TgphComponentProps<typeof Combobox.Options>;
+  multiple?: boolean;
+} & (
+    | {
+        multiple?: true;
+        value?: MultipleValue;
+        onValueChange?: (value: MultipleValue) => void;
+        layout?: TgphComponentProps<typeof Combobox.Root>["layout"];
+      }
+    | {
+        multiple?: false | undefined;
+        value?: SingleValue;
+        onValueChange?: (value: SingleValue) => void;
+        layout?: never;
+      }
+  );
+
+const Root = ({
+  size = "1",
+  multiple = false,
+  value,
+  onValueChange,
+  triggerProps,
+  contentProps,
+  optionsProps,
+  children,
+  ...props
+}: RootProps) => {
+  const [options, setOptions] = React.useState<Array<Option>>([]);
```

Comment:
Is there an issue that occurs if you assign `options` without using `useState` and `useEffect`? I’m wondering if you can just move all this logic into an `options = useMemo(() => ...)`.

Reply:
Yeah good call 👍